front/src/modules/timesStops/hooks/useComputedMargins.tsx

```diff
@@ -0,0 +1,73 @@
+import type { SimulationResponseSuccess } from 'applications/operationalStudies/types';
+import type { TrainScheduleResult } from 'common/api/generatedEditoastApi';
+import type { OperationalPointWithTimeAndSpeed } from 'modules/trainschedule/components/DriverTrainScheduleV2/types';
+import { interpolateValue } from 'modules/trainschedule/components/DriverTrainScheduleV2/utils';
+
+import { marginRegExValidation } from '../consts';
+import { formatDigitsAndUnit } from '../utils';
+
+let marginUnit = '%';
+
+const useComputedMargins = (
+  trainSimulation: SimulationResponseSuccess,
+  opPoint: OperationalPointWithTimeAndSpeed,
+  nextOpPoint: OperationalPointWithTimeAndSpeed,
+  selectedTrainSchedule: TrainScheduleResult,
+  pathId: string
+) => {
+  const theoreticalMarginIndex = selectedTrainSchedule?.margins?.boundaries?.findIndex(
+    (id) => id === pathId
+  );
+  if (!theoreticalMarginIndex) {
+    return ['', '', ''];
+  }
+  const theoreticalMargin = selectedTrainSchedule.margins!.values[theoreticalMarginIndex + 1];
+  const theoMarginRegexMatches = theoreticalMargin?.match(marginRegExValidation);
+  marginUnit = theoMarginRegexMatches?.[3] || marginUnit;
+  const opPointBaseTime = interpolateValue(trainSimulation.base, opPoint?.position * 1000, 'times');
+  const nextOpPointBaseTime = interpolateValue(
+    trainSimulation.base,
+    nextOpPoint?.position * 1000,
+    'times'
+  );
+  const baseDuration = nextOpPointBaseTime - opPointBaseTime;
+  const opPointprovTime = interpolateValue(
```

Comment:
Use the full word here, since it is not a word we see everyday
```suggestion
  const opPointprovisionalTime = interpolateValue(
```

Reply:
done for both opPointprovisionalTime nextOpPointProvisionalTime